cmake/checks/check_01_cxx_features.cmake

```
@@ -347,13 +349,27 @@ if(NOT DEAL_II_HAVE_CXX14)
     "C++14 support. Make sure to use a modern enough compiler (GCC version "
     "5 onwards, Clang version 4 onwards, or Microsoft MS VS 2015 onwards) "
     "and check that the compiler flag \"-std=\" is either unset, or set to "
-    "at least c++14.\n\n"
+    "at least c++14. Similarly, please make sure that the CMake variable "
+    "CMAKE_CXX_STANDARD is either unset, or set at least to 14.\n\n"
     )
 endif()
 
 _test_cxx17_support()
 _test_cxx20_support()
 
+set_if_empty(CMAKE_CXX_STANDARD "${_cxx_standard}")
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+if(NOT "${CMAKE_CXX_STANDARD}" STREQUAL "${_cxx_standard}")
```

Could we run into a case where the feature test for a higher standard would still succeed although the standard flag passed implicitly is for a lower standard?
```suggestion
if("${CMAKE_CXX_STANDARD}" LESS "${_cxx_standard}")
```

---

If checked that we cannot:

if you set `CMAKE_CXX_STANDARD` then CMake will issue `-std=c++??` (when necessary) on the command line for all compilation tests. Thus, `CMAKE_CXX_STANDARD` will always be greater or equal to `_cxx_standard`.

---

Yeah, I wasn't quite sure if a compiler would still compile a newer feature if an older `-std=c++*` flag is set but trying for my self, this doesn't seem to be the case.